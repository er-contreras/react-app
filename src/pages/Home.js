import React from 'react';
import { LoremIpsum } from 'react-lorem-ipsum';

const Home = () => { // eslint-disable-line
  return (
    <div
      style={{
        fontSize: '1rem',
        margin: '2rem 0',
      }}
    >
      <h2
        style={{
          margin: '2rem 0',
        }}
      >
        Welcome to our page!
      </h2>
      <LoremIpsum p={2} />
    </div>
  );
};

export default Home;
