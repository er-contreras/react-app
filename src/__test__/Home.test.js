import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from '../pages/Home';

describe('Home test', () => {
  it('Should render same text passed into ...', () => {
    const tree = TestRenderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the main heading', () => {
    render(<Home />);
    const heading = screen.getByText(/Welcome to our page!/i);
    expect(heading).toMatchSnapshot();
  });
});
