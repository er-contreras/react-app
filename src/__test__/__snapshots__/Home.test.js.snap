// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Should render same text passed into ... 1`] = `
<div
  style={
    Object {
      "fontSize": "1rem",
      "margin": "2rem 0",
    }
  }
>
  <h2
    style={
      Object {
        "margin": "2rem 0",
      }
    }
  >
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum odor amet, consectetuer adipiscing elit. Purus quis posuere erat erat libero mattis tellus dui pretium. Gravida lobortis tellus ultricies ultrices sagittis montes primis. Facilisis eu aptent est et class vehicula euismod. Lorem senectus suspendisse sapien nec ligula interdum dignissim vehicula. Fringilla cursus convallis enim eleifend facilisi suscipit ornare scelerisque. In finibus donec accumsan mollis nullam congue. Nisl consequat feugiat commodo blandit dui tristique blandit suspendisse. Habitasse leo ipsum amet etiam nostra semper leo ornare vel.
  </p>
  <p>
    Lorem ipsum odor amet, consectetuer adipiscing elit. Purus quis posuere erat erat libero mattis tellus dui pretium. Gravida lobortis tellus ultricies ultrices sagittis montes primis. Facilisis eu aptent est et class vehicula euismod. Lorem senectus suspendisse sapien nec ligula interdum dignissim vehicula. Fringilla cursus convallis enim eleifend facilisi suscipit ornare scelerisque. In finibus donec accumsan mollis nullam congue. Nisl consequat feugiat commodo blandit dui tristique blandit suspendisse. Habitasse leo ipsum amet etiam nostra semper leo ornare vel.
  </p>
</div>
`;

exports[`Home test renders the main heading 1`] = `
<h2
  style="margin: 2rem 0px;"
>
  Welcome to our page!
</h2>
`;
